labs/lab-2-javascript-game/elicecilgame.js

```diff
@@ -0,0 +1,102 @@
+class Player {
+    constructor(name, health, healsRemaining, wins) {
+        this.name = name;
+        this.health = health;
+        this.healsRemaining = healsRemaining;
+        this.wins = wins;
+        this.maxHealth = health;
+    }
+
+    generateAttackDamage() {
+        var damage = Math.round(Math.random() * 3);      
+        return damage;
+    }
+
+    hitPlayer(damage) {
+        this.health = this.health - damage;
+    }
+
+    heal() {
+        var heal = Math.round(Math.random() * 10);  
+        return heal;   
+    }
+
+    healPlayer(heal) {
+        this.health = this.health + heal;
+        this.healsRemaining = this.healsRemaining - 1;   
+    }    
+}
+
+
+function startGame() {
+    var attack = true;
+    var start = confirm("Would you like to fight the Champion?");
+    var userName = prompt("Tell us your name."); 
+    
+    var user = new Player(userName, 40, 2, 0);
+    const grant = new Player("Grant", 10, 0, 0);
+
+    function updateHealthBar(player, healthbar) {
+        if(player.health >= player.MaxHealth) {
+            healthbar.style.width = "200px";
+        } else {
+            var barWidth = (200 / player.maxHealth) * (player.health);
+            healthbar.style.width = barWidth + "px";            
+        }
+    }
+
+    function updateHealth(user, grant) {
+        var userHealthBar = document.getElementById("userhealthbar");
+        var enemyHealthBar = document.getElementById("enemyhealthbar");
+
+        document.getElementById("healthcounter").innerText = user.name + " has " + user.health + " health left. Grant has " + grant.health + " health left. ";
+        updateHealthBar(user, userHealthBar);
+        updateHealthBar(grant, enemyHealthBar);
+    }
+    
+    document.getElementById("statsName").innerText = userName;
```

Comment:
This is just kinda hanging out here, in the middle of a bunch of functions. A surprising place to put it, and surprises are bad.